### ISSUE TITLE
Gradle build is broken since 4.5.3 when using wire.protoLibrary = true

### ISSUE THREAD

**Issue by sergei-lapin** [square/wire#2440]
Since 4.5.3 Gradle modules configured to be proto libs via `wire.protoLibrary = true` are destined to raise warning (older Gradle versions) or fail build (newer Gradle versions) due to the undeclared task dependencies:

```
Reason: Task ':processResources' uses this output of task ':generateMainProtos' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
```

[Here](https://github.com/sergei-lapin/issue-reproducers/tree/wire-gradle-issues) you can find a MRE (commit f97d92c)

This is reproducible with Gradle at least 7.6.1 (tested with 7.6.1 and 8.1.1)

**Comment by oldergod** [comment]
What's the step to see the reason showing somewhere?

**Comment by sergei-lapin** [comment]
closing as it was resolved with the release of 4.7.0
